Fix test in PR 12663

XD-DENG opened the pull request:
The order of the two TI keys in `test_process_task_instances_depends_on_past()` (https://github.com/apache/airflow/pull/12663) is not guaranteed,
in such a case, directly assert two lists (both with two elements) will cause 'random' failure

This PR addresses this issue.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

XD-DENG commented on tests/jobs/test_scheduler_job.py:
```diff
@@ -516,12 +516,12 @@ def test_process_task_instances_with_task_concurrency(
+        ti_to_schedule.append.assert_called_once_with(
             (dag.dag_id, dag_task1.task_id, DEFAULT_DATE, TRY_NUMBER),
-        ]
+        )
```
We don't "have to" make change for this line, given for this case we only have one TI (i.e. one element in the list). 

Changing only for consistent testing style/method.

kaxil commented on tests/jobs/test_scheduler_job.py:
```diff
@@ -567,13 +567,15 @@ def test_process_task_instances_depends_on_past(self, state, start_date, end_dat
-            (dag.dag_id, dag_task1.task_id, DEFAULT_DATE, TRY_NUMBER),
-            (dag.dag_id, dag_task2.task_id, DEFAULT_DATE, TRY_NUMBER),
-        ]
+        ti_to_schedule.append.assert_called_with(
```
Maybe just make the list sorted:

```
assert sorted([(dag.dag_id, dag_task1.task_id, DEFAULT_DATE, TRY_NUMBER), (dag.dag_id, dag_task2.task_id, DEFAULT_DATE, TRY_NUMBER),]) == sorted(ti_to_schedule)
```

XD-DENG replied:
Makes sense, and the length can be asserted too in this way. Why not.

Re-pushed.

XD-DENG committed:
Fix test in PR 12663

The order of the two TI keys in test_process_task_instances_depends_on_past() is not guaranteed,
in such a case, directly assert two lists (both with two elements) will cause 'random' failure

This PR addresses this issue.

Co-authored-by: Kaxil Naik <kaxilnaik@gmail.com>

potiuk reviewed (approved):
Oh yeah!
